gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2

```diff
@@ -180,10 +186,8 @@ class {{ service.name }}RestTransport({{ service.name }}Transport):
         # TODO(yon-mg): further discussion needed whether 'python truthiness' is appropriate here
         #               discards default values
         # TODO(yon-mg): add test for proper url encoded strings
-        query_params = ((k, v) for k, v in query_params.items() if v)
-        for i, (param_name, param_value) in enumerate(query_params):
-            q = '?' if i == 0 else '&'
-            url += "{q}{name}={value}".format(q=q, name=param_name, value=param_value.replace(' ', '+'))
+        query_params = ['{k}={v}'.format(k=k, v=v) for k, v in query_params.items() if v]
+        url += '?{}'.format('&'.join(query_params)).replace(' ', '+')
```

Comment:
It might be easier to use `urllib.parse.urlencode` here (It will create the `k=v` structure and glue the pairs together with `&` characters).

https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode
https://docs.python.org/3/library/urllib.request.html#urllib-examples

Reply:
Not sure how I missed this. Thanks!

Reply:
To clarify, I'll leave this out for now so I don't accidentally break anything but I'll update in the subsequent PR